genesis/src/lib.rs

```diff
@@ -27,12 +29,81 @@ pub static GENESIS_ACCOUNT_ID: Lazy<AccountId> =
     Lazy::new(|| AccountId::new(GENESIS_DOMAIN_ID.clone(), "genesis".parse().expect("Valid")));
 
 /// Genesis transaction
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Decode, Encode, Deserialize, Serialize)]
 #[repr(transparent)]
 pub struct GenesisTransaction(pub SignedTransaction);
 
+impl GenesisTransaction {
+    /// Construct single genesis transaction with [`Executor`] and transactions unified under a single transaction
+    pub fn new_unified(
+        raw_block: RawGenesisBlock,
+        chain_id: &ChainId,
+        genesis_key_pair: &KeyPair,
+    ) -> GenesisTransaction {
+        Self(raw_block.unify().sign(chain_id.clone(), genesis_key_pair))
+    }
+}
+
+/// [`SignedGenesisConfig`] contains data that is used for loading signed genesis from config.
+#[derive(Debug, Clone, Deserialize, Serialize, Decode, Encode)]
+pub struct SignedGenesisConfig {
+    chain_id: ChainId,
+    creation_time_ms: u64,
+    signatures: SignaturesOf<TransactionPayload>,
+}
+
+impl SignedGenesisConfig {
+    /// Create [`SignedGenesisConfig`] from it's components
+    pub fn new(
+        chain_id: ChainId,
+        creation_time_ms: u64,
+        signatures: SignaturesOf<TransactionPayload>,
+    ) -> Self {
+        Self {
+            chain_id,
+            creation_time_ms,
+            signatures,
+        }
+    }
+
+    /// Checks that [`SignedGenesisConfig`] corresponds to [`RawGenesisBlock`] and produces [`GenesisNetwork`] if it does.
+    /// # Errors
+    /// Fails if [`RawGenesisBlock`] does not correspond to [`SignedGenesisConfig`] and it was unable to verify it's integrity
+    pub fn validate(self, genesis_block: RawGenesisBlock) -> Result<GenesisNetwork> {
```

Comment:
This function seems to belong more to `GenesisNetwork` itself:

```rust
impl GenesisNetwork {
    fn try_parse(raw_block: RawGenesisBlock, signature: RawGenesisBlockSignature) -> Result<Self> {}
}
```

Reply:
Also, if we stick to the article ["Parse, don't validate"](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/), this is the case of __parse__.

Reply:
Thank you for the read, I agree with you on that!